rocAL/rocAL_pybind/examples/rocAL_api_external_source_example.py

```diff
@@ -0,0 +1,80 @@
+import types
+import collections
+import numpy as np
+from random import shuffle
+from amd.rocal.pipeline import Pipeline
+from amd.rocal.plugin.pytorch import ROCALClassificationIterator
+import amd.rocal.fn as fn
+import amd.rocal.types as types
+
+def main():
+
+    batch_size = 3
+    data_dir = "/media/MIVisionX-data/rocal_data/coco/coco_10_img/train_10images_2017/" # Pass the data directory
+    device = "cpu"
+
+    def draw_patches(img, idx, device):
+    #image is expected as a tensor, bboxes as numpy
+        import cv2
+        image = img.detach().numpy()
+        image = image.transpose([1, 2, 0]) # NCHW
+        image = (image).astype('uint8')
+        image = cv2.cvtColor(image, cv2.COLOR_RGB2BGR)
+        cv2.imwrite(str(idx)+"_"+"train"+".png", image)
+
+    #Define the Data Source for all image samples - User needs to define their own source
+    class ExternalInputIteratorMode0(object):
+        def __init__(self, batch_size):
+            self.images_dir = data_dir
+            self.batch_size = batch_size
+            self.files = []
+            import os, glob
+            for filename in glob.glob(os.path.join(self.images_dir, '*.jpg')):
+                self.files.append(filename)
+            shuffle(self.files)
+
+        def __iter__(self):
+            self.i = 0
+            self.n = len(self.files)
+            return self
+
+        def __next__(self):
+            batch = []
+            label = []
+            for i in range(self.batch_size):
+                jpeg_filename = self.files[self.i]
+                batch.append(jpeg_filename)
+                label.append(1) # Label is some random variable for now - iser can modify acording to use case
+                self.i = (self.i + 1) % self.n
+            return batch, label
+    
+    # Mode 0
+    eii = ExternalInputIteratorMode0(batch_size)
+
+    #Create the pipeline 
+    external_source_pipeline_mode0 = Pipeline(batch_size=batch_size, num_threads=1, device_id=0, seed=1, rocal_cpu=True, tensor_layout=types.NCHW , tensor_dtype=types.FLOAT)
```

Comment:
Please add space before and after = operator and check the spacing before comma

Reply:
@swetha097 Please check this